elixir/apps/domain/lib/domain/notifications/jobs/outdated_gateways.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Domain.Notifications.Jobs.OutdatedGateways do
+  use Domain.Jobs.Job,
+    otp_app: :domain,
+    every: :timer.minutes(5),
+    executor: Domain.Jobs.Executors.GloballyUnique
+
+  require Logger
+  require OpenTelemetry.Tracer
+
+  alias Domain.Actors
+  alias Domain.{Accounts, Gateways, Mailer}
+
+  @impl true
+  if Mix.env() == :prod do
+    def execute(_config) do
+      # Should only run on Sundays
+      day_of_week = Date.utc_today() |> Date.day_of_week()
+      if day_of_week == 7, do: run_check()
+    end
+  else
+    def execute(_config) do
+      run_check()
+    end
+  end
+
+  defp run_check do
+    Accounts.all_active_paid_accounts!()
```

Comment:
I think it would be better to change the query function to do what `nofied_today?` does so that we don't need to fetch all paid accounts every time:
```suggestion
    Accounts.all_active_paid_accounts_pending_notification!(:outdated_gateways)
```

Reply:
Yeah, in theory I agree, but in practice do you think it will actually make any difference?

Along with that, if I were to move this to the SQL query rather than filtering on the Elixir side, wouldn't I have to guarantee that the `last_notified` field has a timestamp and not a `null` value?  Right now we don't set any type of default value in the DB for the `last_notified` value.

Or were you suggesting doing the filtering in Elixir, just inside the `all_active_paid_accounts_pending_notification` function?

Reply:
It will make difference because it might become a time bomb if we get more rapid growth, at some point loading all accounts in memory will crash the jobs runner. With a filter it's still possible but far more likely. I was thinking just to add something like this to the `all_active_paid_accounts_pending_notification!`:

```elixir
where(
      queryable,
      [accounts: accounts],
      fragment("(?->'notifications'->?->'last_notified')::timestamp < NOW() - interval '24 hours'", accounts.config, :outdated_gateway, ^datetime)
    )
```

it should handle `NULL` by itself because a `NULL` in a `WHERE` clause will be processed as `FALSE`:
```
SELECT  NOW();
-- 2024-10-09 18:08:59.79317+00

SELECT NOW() WHERE NULL;
-- 0 rows

SELECT 1 WHERE NULL::timestamp < NOW() - interval '24 hours';
-- 0 rows

SELECT 1 WHERE '2024-10-09 18:08:59.79317+00'::timestamp < NOW() - interval '24 hours';
-- 0 rows

SELECT 1 WHERE '2024-10-08 10:08:59.79317+00'::timestamp < NOW() - interval '24 hours';
-- 1 row

```